src/components/UserList/UserList.tsx

```diff
@@ -0,0 +1,197 @@
+  return order === Order.DESC
+    ? (firstValue, secondValue) => {
+        return descendingComparator(firstValue, secondValue, orderBy);
+      }
+    : (firstValue, secondValue) => {
+        return -descendingComparator(firstValue, secondValue, orderBy);
+      };
+}
+
+export function UserList() {
+  const [order, setOrder] = useState<Order>(Order.ASC);
+  const [orderBy, setOrderBy] = useState<keyof Data>('id');
+  const [selected, setSelected] = useState<readonly number[]>([]);
+  const [page, setPage] = useState(0);
+  const [rows, setRows] = useState<Data[]>([]);
+  const defaultRowsPerPage = 10;
+  const [rowsPerPage, setRowsPerPage] = useState(defaultRowsPerPage);
+  const pixelHeightPerRow = 53;
+  const fivePerPage = 5;
+  const tenPerPage = 10;
+  const twentyFivePerPage = 25;
+
+  const handleRequestSort = (event: MouseEvent<unknown>, property: keyof Data) => {
+    const isAsc = orderBy === property && order === Order.ASC;
+    setOrder(isAsc ? Order.DESC : Order.ASC);
+    setOrderBy(property);
+  };
+
+  const handleSelectAllClick = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const newSelected: readonly number[] = rows.map((n: { id: number }) => {
+        return n.id;
+      });
+      setSelected(newSelected);
+      return;
+    }
+    setSelected([]);
+  };
+
+  const handleClick = (event: MouseEvent<unknown>, id: number) => {
+    const selectedIndex = selected.indexOf(id);
+    let newSelected: readonly number[] = [];
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, id);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(selected.slice(0, selectedIndex), selected.slice(selectedIndex + 1));
+    }
+
+    setSelected(newSelected);
+  };
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const isSelected = (id: number) => {
+    return selected.indexOf(id) !== -1;
+  };
+
+  const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - rows.length) : 0;
+
+  class FetchError extends Error {
+    constructor(message: string) {
+      super(message);
+      this.name = 'FetchError';
+    }
+  }
+
+  useEffect(() => {
+    fetch('../src/components/UserList/testUsers.json')
+      .then((res) => {
+        return res.json();
+      })
+      .then((data: Data[]) => {
+        return setRows(data);
+      })
+      .catch((error) => {
+        if (error instanceof FetchError) {
+          throw new FetchError(`Failed to fetch users with error: ${error.message}`);
+        }
+      });
+  });
```

Comment:
now you will trigger data fetch every time you will make any changes to this component this is something bad, but as it is just a mock data that we are going to replace soon this can stay